Client/Frontend/Reader/ReaderMode.swift

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+
+enum ReaderModeState: String {
+    case Available = "Available"
+    case Unavailable = "Unavailable"
+    case Active = "Active"
+}
+
+/// Delegate that contains callbacks that we have added on top of the built-in WKWebViewDelegate
+protocol ReaderModeDelegate {
+    func readerMode(readerMode: ReaderMode, didChangeReaderModeState state: ReaderModeState, forBrowser browser: Browser)
+}
+
+class ReaderMode: BrowserHelper {
+    var delegate: ReaderModeDelegate?
```

Comment:
I think it's generally a good idea to make delegates weak -- that's what iOS does everywhere. I don't think you have a ref cycle in this case, but it'd be a good change anyway for the sake of future proofing.